genref/config.yaml

```diff
@@ -0,0 +1,139 @@
+    package: k8s.io/kubelet
+    path: config/v1beta1
+    includes:
+      - k8s.io/component-base/config/v1alpha1
+
+  - name: kubelet-stats
+    package: k8s.io/kubelet
+    path: pkg/apis/stats/v1alpha1
+    # This is skipped because the 'stats' API is not yet available in 0.19.3
+    skip: true
+
+  - name: kube-scheduler-policy-config
+    package: k8s.io/kube-scheduler
+    path: config/v1
+
+  - name: kube-scheduler-config
+    package: k8s.io/kube-scheduler
+    path: config/v1beta1
+
+  - name: kube-controller-manager-config
+    package: k8s.io/kube-controller-manager
+    path: config/v1alpha1
+    includes:
+      - k8s.io/cloud-provider/controllers/service/config/v1alpha1
+      - k8s.io/cloud-provider/config/v1alpha1
+      - k8s.io/controller-manager/config/v1alpha1
+
+  - name: kube-proxy-config
+    package: k8s.io/kube-proxy
+    path: config/v1alpha1
+
+  - name: metrics
+    package: k8s.io/metrics
+    path: pkg/apis/metrics/v1alpha1
+
+  - name: metrics
+    package: k8s.io/metrics
+    path: pkg/apis/metrics/v1beta1
+
+  - name: client-authentication
+    package: k8s.io/client-go
+    path: pkg/apis/clientauthentication/v1alpha1
+
+  - name: client-authentication
+    package: k8s.io/client-go
+    path: pkg/apis/clientauthentication/v1beta1
+
+  - name: apiserver-config
+    package: k8s.io/apiserver
+    path: pkg/apis/apiserver/v1alpha1
+
+  # This only exists in v0.19.3
+  - name: apiserver-config
+    package: k8s.io/apiserver
+    path: pkg/apis/apiserver/v1beta1
+    skip: true
+
+  - name: apiserver-config
+    package: k8s.io/apiserver
+    path: pkg/apis/apiserver/v1
+
+  - name: apiserver-encryption
+    package: k8s.io/apiserver
+    path: pkg/apis/config/v1
+
+  - name: apiserver-audit
+    package: k8s.io/apiserver
+    path: pkg/apis/audit/v1
+
+  - name: apiserver-webhookadmission
+    package: k8s.io/apiserver
+    path: pkg/admission/plugin/webhook/config/apis/webhookadmission/v1alpha1
+
+  - name: apiserver-webhookadmission
+    package: k8s.io/apiserver
+    path: pkg/admission/plugin/webhook/config/apis/webhookadmission/v1
+
+  - name: apiserver-eventratelimit
+    package: k8s.io/kubernetes
+    path: plugin/pkg/admission/eventratelimit/apis/eventratelimit/v1alpha1
+    # Skipped because k8s.io/kubernetes cannot be imported
+    skip: true
+
+  - name: apiserver-podtolerationrestriction
+    package: k8s.io/kubernetes
+    path: plugin/pkg/admission/podtolerationrestriction/apis/podtolerationrestriction/v1alpha1
+    # Skipped because k8s.io/kubernetes cannot be imported
+    skip: true
+
+  - name: cloud-provider-config
+    package: k8s.io/cloud-provider
+    path: config/v1alpha1
+    includes:
+      - k8s.io/controller-manager/config/v1alpha1
+      - k8s.io/cloud-provider/controllers/service/config/v1alpha1
+
+  - name: cloud-provider-service
+    package: k8s.io/cloud-provider
+    path: controllers/service/config/v1alpha1
+
+  # This is for testing
+  - name: kubeadm-config
```

Comment:
Is there a reason to include the `kubeadm-config` entries as "tests"?

Reply:
kubeadm config is a very important configuration for users to deploy and upgrade a cluster. Documenting kubeadm config, however, is not straightforward because the "API" resides in the kubeadm cmd package. It has many dependencies to other files in the k8s.io/kubernetes repo. What makes things worse is that `k8s.io/kubernetes` itself cannot be imported or vendored. After consulting some kubeadm experts, the temporary (testing) workaround is to clone that API somewhere and then eliminate the dependencies in the clone.

In the long run, kubeadm wants to publish its APIs just like other components do. We will revise the path for these APIs when that refactoring has landed.